### PR TITLE
Selectable shared or static linking of libnccl

### PR DESCRIPTION
**Pull request by nluehr**
By default, nccl is linked statically.
To link against the shared lib `export HOROVOD_NCCL_LINK=SHARED`.

### CODE REVIEW

**Comment by alsrgv** [comment]
@nluehr, thanks for your PR.  Please sign the CLA and we'll get it going.

**Comment by alsrgv** [comment]
@nluehr, any luck with the CLA?

**Comment by nluehr** [comment]
CLA signed.

File: setup.py [hunk 1]
```diff
@@ -323,7 +329,7 @@ def get_common_options(build_ext):
 
     if gpu_allreduce == 'NCCL':
         have_nccl = True
-        nccl_include_dirs, nccl_lib_dirs = get_nccl_dirs(
+        nccl_include_dirs, nccl_lib_dirs, nccl_lib_name = get_nccl_vals(
             build_ext, cuda_include_dirs, cuda_lib_dirs, cpp_flags)
     else:
         have_nccl = False
```
Comment (alsrgv) [inline setup.py]: Can you add `nccl_lib_name = None` after `nccl_include_dirs = nccl_lib_dirs = []` below to remain consistent?  Or, just remove the `nccl_include_dirs = nccl_lib_dirs = []` line.

Comment (alsrgv) [reply setup.py]: Or, even better, return `nccl_libs` from `get_nccl_vals()` which is an array of library names.  Then you can set all three to an empty array, and it's also a nice default value - better than an array wth None element.

**Comment by nluehr** [comment]
Requested change complete.

**Pull request merged**